parsl/providers/slurm/slurm.py

```diff
@@ -237,17 +238,21 @@ def cancel(self, job_ids):
         rets = None
         if retcode == 0:
             for jid in job_ids:
-                self.resources[jid]['status'] = translate_table['CA']  # Setting state to cancelled
+                self.resources[jid]['status'] = JobStatus(JobState.CANCELED)  # Setting state to cancelled
```

Comment:
should be two Ls in this CANCELED constant?

Reply:
Right. Fixed by e117128. I'm a bit surprised that typechecking doesn't catch this. Kinda defeats the purpose of enums.

Reply:
At least in parsl CI, mypy is configured to only typecheck functions which have type signatures (that is mypy default behaviour, and the parsl codebase isn't ready for anything more aggressive).

I added this mostly-no-op type signature to cancel in the slurm provider:

```
-    def cancel(self, job_ids):
+    def cancel(self, job_ids) -> Any:
```

on the commit right before e117128 and mypy does detect it then:

> parsl/providers/slurm/slurm.py:241: error: "Type[JobState]" has no attribute "CANCELED"

I guess the moral of the story is to press on with making type checking deployed more throughout the codebase?

Reply:
Right. I'm not quite sure why adding type annotations is such a piecewise effort. It seems a bit wasteful to me.